src/content/basics/cluster-size-autoscaling/index.md

```diff
@@ -0,0 +1,66 @@
++++
+title = "Cluster Size and Autoscaling"
+description = "This article explains options you have for defining the size of a Kubernetes cluster with Giant Swarm, and automatically scaling it"
+date = "2019-02-01"
+weight = 45
+type = "page"
+categories = ["basics"]
++++
+
+# Cluster Size and Autoscaling
+
+Starting with release version {{% first_aws_autoscaling_version %}} for AWS, you can leverage the benefits of the [Kubernetes autoscaler](https://github.com/kubernetes/autoscaler) to define the number of worker nodes in a cluster based on demand.
+The autoscaler is provided with every cluster of version >= {{% first_aws_autoscaling_version %}} on AWS.
+
+On Giant Swarm installations on Azure, on bare-metal, and on AWS prior to version {{% first_aws_autoscaling_version %}}, the cluster size would be defined statically.
+
+## Setting scaling limits
+
+With the autoscaler taking control over the number of worker nodes, you only define the bounds or limits in which the cluster size can vary. This is possible both during cluster creation as well as after creation.
+
+To enforce an exact cluster size and **effectively disable the autoscaler**, simply set the minimum and maximum worker node count to the same value. This is also the recommended way in case you want to control the number of worker nodes through some external tooling via the Giant Swarm API.
+
+## Minimal and default cluster size
+
+When creating a cluster without specifying the number of worker nodes, {{% default_cluster_size_worker_nodes %}} worker nodes will be created. This is also the minimal number supported by Giant Swarm. On AWS starting with release version {{% first_aws_autoscaling_version %}}, when not specified, the maximum number of worker nodes is also set to {{% default_cluster_size_worker_nodes %}}.
+
+Technically, while you may be able to create and run smaller clusters successfully, we don't encourage this due to reduced resilience. We explicitly deactivate all sorts of alerts for clusters with less than {{% minimal_supported_cluster_size_worker_nodes %}} worker nodes and won't get notified in case of any problems.
+
+## Minimal worker node instance type
+
+Relying on autoscaling may have an effect on the instance type you should use for your cluster.
+To illustrate this more detail, it is helpful to explain how the autoscaler determines whether a node is needed or not.
+
+The autoscaler periodically calculates the utilization of a node based on CPU and memory requests of all running pods, compared to the node's total capacity.
+
+**Note:** this means that your workloads need sensible CPU and memory requests in order to inform the autoscaler about the actual node utilization.
+
+The current utilization is compared to a configurable _utilization threshold_, which is {{% autoscaler_utilization_threshold %}} by default.
+If the utilization is below the threshold, the autoscaler decides to remove the node.
+If it's above, it will keep the node.
```

Comment:
Here we describe what conditions need to be met in order to scale down, but we don't write anywhere about conditions that need to be met for scaling up.  It might even be worth separate paragraph so this info can be easily found without the need to read every word and I personally would never think to look for that info in `Minimal worker node instance type` section

Reply:
@calvix I agree that the explanation on how the autoscaler works should be moved into its own section. Will cahnge.

@teemow There are several instances types in our supported list that might be too small for the default utilization threshold. c5.large, m3.large, m4.large, m5.large, r3.large, t3.large all have 2 cores only. And this list may vary. Also, our component's footprint may vary. So I wonder how we can keep this information useful AND maintainable. But for now I'd be fine with a list of instance types to stay away from.

Reply:
@teemow BTW have you seen that a bit further down the text contains `We recommend to use types with at least 4 CPU cores for autoscaling clusters.` ?

Reply:
> have you seen that a bit further down the text

Yes that's why I said it would be nice if we would name instance types instead of how many CPUs. Nobody knows this without looking it up.

Reply:
Considering this resolved. We tested with `c5.large`, which is the smallest type from our supported list, and scaling down worked. The information in the text should be sufficient.